### ISSUE TITLE
Emit error when missing comma

### ISSUE THREAD

**Issue by dragon-archer** [RedCMD/TmLanguage-Syntax-Highlighter#9]
![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/45531248/d5020bbd-f244-4799-9c63-ba53b97c332e)

Currently, this extension will silently accept items without comma

![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/45531248/d6e92155-d606-4249-9916-8f805111dfd9)

while the builtin`json` mode will show a prominent error

How about implementing this by using a simple sementic highlight or some textmate rules like:
```json
{
    "match": "\\}\\s*\\{",
    "name": "invalid.illegal.xxx"
}
```

Or perhaps you'd like to wait for the official VS Code Team to support using custom textmate rules together with the builtin mode together (As I've seen in your other issues)

**Comment by RedCMD** [comment]
yeah, same as in #3
I'm waiting on VSCode to allow custom extensions to enable the builtin json extension
https://github.com/microsoft/vscode/issues/198025
https://github.com/microsoft/vscode/pull/198583

if you want
there is code ready for displaying Treesitter parse errors
you'll just need to import the file and enable it
it wont show what type of error, other than it being an error
(reports on false postive json errors is welcomed)

https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/blob/bcb5d1045fb14393be3a30a5c7320b8e9315c868/out/extension.js#L5-L6
`const DiagnosticCollection_1 = require("./DiagnosticCollection");`
![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/33529441/1665a17f-73af-4baf-969e-ec808fc615a8)

https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/blob/bcb5d1045fb14393be3a30a5c7320b8e9315c868/out/extension.js#L20 `(0, DiagnosticCollection_1.initDiagnostics)(context);`
![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/33529441/3190c714-cd42-4b99-a897-fe09142c1831)

https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/blob/bcb5d1045fb14393be3a30a5c7320b8e9315c868/out/DiagnosticCollection.js#L33
and prob disable the regex errors (many false postives)
https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/blob/bcb5d1045fb14393be3a30a5c7320b8e9315c868/out/DiagnosticCollection.js#L49

I could look into making it a config option to enable/disable error reporting
tho I'll prob have it disabled by default for now

**Comment by RedCMD** [comment]
or the other way is manually enabling the builtin `json-language-features` extension

`\extensions\json-language-features\client\dist\node\jsonClientMain.js`
just involves adding the `languageId` to the document selector list `"jsonc","json-textmate"`
![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/33529441/804e968c-26c4-4298-a5e7-192e7d981c59)

and `"activationEvents"`
`\extensions\json-language-features\package.js`
`"onLanguage:json-textmate"`
![image](https://github.com/RedCMD/TmLanguage-Syntax-Highlighter/assets/33529441/5a676b25-0c34-470a-9b28-06de17d541c6)

**Comment by dragon-archer** [comment]
Thanks, I turned on the builtin feature, though I think it might be better with a config option in this extension

**Comment by RedCMD** [comment]
Now available in the full release 2.3.0 for VSCode 1.87.0
